esmvaltool/preprocessor/_multimodel.py

```diff
@@ -59,62 +317,72 @@ def multi_model_mean(cubes, span, filename, exclude):
         return cubes
 
     # check if we have any time overlap
-    if _get_overlap(cubes) is None:
+    ovlp = _get_overlap(selection)
+    if ovlp is None:
         logger.info("Time overlap between cubes is none or a single point.")
         logger.info("check models: will not compute statistics.")
         return cubes
-    else:
-        # empty lists to hold data
-        tmeans = []
-        data_size = []
-
-        # add file name info
-        file_names = [
-            os.path.basename(cube.attributes.get('_filename'))
-            for cube in cubes
-        ]
-
-        # look at overlap type
-        if span == 'overlap':
-            tx1, tx2 = _get_overlap(cubes)
-            for cube in selection:
-                logger.debug("Using common time overlap between "
-                             "models to compute statistics.")
-                logger.debug("Bounds: %s and %s", str(tx1), str(tx2))
-                # slice cube on time
-                with iris.FUTURE.context(cell_datetime_objects=True):
-                    cube = _slice_cube(cube, tx1, tx2)
-
-                # store nr time points
-                data_size.append(len(list(cube.coord('time').points)))
-
-                # compute stats and append to lists
-                tmeans.append(cube.collapsed('time', iris.analysis.MEAN))
-
-        elif span == 'full':
-            for cube in selection:
-                logger.debug("Using full time spans "
-                             "to compute statistics.")
-
-                # store nr time points
-                data_size.append(len(list(cube.coord('time').points)))
-
-                # compute stats and append to lists
-                tmeans.append(cube.collapsed('time', iris.analysis.MEAN))
 
-        else:
-            logger.debug("No type of time overlap specified "
-                         "- will not compute cubes statistics")
-            return cubes
+    # cases
```

Comment:
This section still looks like 2 times almost the same code. Try to rewrite it in such a way that there is no code duplication.